src/svm/svc.rs

```diff
@@ -184,8 +178,8 @@ impl<'a, TX: Number + RealNumber, TY: Number + Ord, X: Array2<TX>, Y: Array1<TY>
         self
     }
     /// The kernel function.
-    pub fn with_kernel(mut self, kernel: &'a (dyn Kernel<'a>)) -> Self {
-        self.kernel = Some(kernel);
+    pub fn with_kernel<K: Kernel + 'static>(mut self, kernel: K) -> Self {
```

Comment:
this makes the kernel object to live for the entire execution of the program: https://doc.rust-lang.org/rust-by-example/scope/lifetime/static_lifetime.html#reference-lifetime

Reply:
I think that at this case it means this: https://doc.rust-lang.org/rust-by-example/scope/lifetime/static_lifetime.html#trait-bound.

```As a trait bound, it means the type does not contain any non-static references```

Reply:
Ah right. lifetime have different syntax than traits, for which can be `!Sized` or also `?Sized`.

Ok then if we are sure that the kernel has right lifetime (e.g. is dropped when assigned variable is out of scope) this can work as expected.